From b81929fa74f29452f6fe874df4b317264f26e89c Mon Sep 17 00:00:00 2001
From: OpenWRT <test@example.com>
Date: Thu, 30 May 2013 16:18:05 +0400
Subject: [PATCH 722/859] 110-fix_mtd_include

---
 include/linux/mtd/physmap.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/include/linux/mtd/physmap.h b/include/linux/mtd/physmap.h
index d2887e7..3eb3143 100644
--- a/include/linux/mtd/physmap.h
+++ b/include/linux/mtd/physmap.h
@@ -17,6 +17,7 @@
 
 #include <linux/mtd/mtd.h>
 #include <linux/mtd/partitions.h>
+#include <linux/platform_device.h>
 
 struct map_info;
 struct platform_device;
-- 
1.8.1.4

